src/tape_drivers/linux/sg/sg_tape.c

```diff
@@ -118,31 +121,25 @@ static inline int _parse_logPage(const unsigned char *logdata,
 								 const uint16_t param, uint32_t *param_size,
 								 unsigned char *buf, const size_t bufsize)
 {
-	uint16_t page_len, param_code, param_len;
-	uint32_t i;
+	const uint16_t page_len = ((uint16_t)logdata[2] << 8) | (uint8_t)logdata[3];
```

Comment:
Need to move to line 128.

And I cannot understand why you change the cast from `uint16_t` to `uint8_t`. Do you have a problem on the precious code?